src/h5web/dataset-visualizer/DatasetVisualizer.tsx

```diff
@@ -1,63 +1,26 @@
-import React, { ReactElement, useEffect, useContext } from 'react';
-import { useSetState, useTimeoutFn } from 'react-use';
-import type { HDF5Dataset, HDF5Value } from '../providers/models';
+import React, { ReactElement, useState, memo } from 'react';
+import type { HDF5Dataset } from '../providers/models';
 import styles from '../Visualizer.module.css';
 import VisSelector from './VisSelector';
 import VisDisplay from './VisDisplay';
-import { DimensionMapping, Vis } from './models';
-import { VIS_DEFS } from '../visualizations';
-import { ProviderContext } from '../providers/context';
+import { Vis } from './models';
 import { useSupportedVis } from './hooks';
+import { VIS_DEFS } from '../visualizations';
 
 interface Props {
   dataset: HDF5Dataset;
 }
 
-interface State {
-  activeVis: Vis;
-  value?: HDF5Value;
-  loading?: boolean;
-  mapperState?: DimensionMapping;
-  prevDataset?: HDF5Dataset;
```

Comment:
The derived state condition below is much simpler, and we no longer need to store the `prevDataset`.